doc/specs/#2046 - Unified keybindings and commands, and synthesized action names.md

```diff
@@ -0,0 +1,589 @@
+default keybindings and these changes is given in [**Appendix
+2**](#appendix-2-complete-default-command-palette).
+
+## Concerns
+
+**TODO: FOR DISCUSSION**: "New tab with index {index}". How does this play with
+the new tab dropdown customizations in [#5888]? In recent iterations of that
+spec, we changed the meaning of `{ "action": "newTab", "index": 1 }` to mean
+"open the first entry in the new tab menu". If that's a profile, then we'll open
+a new tab with it. If it's an action, we'll perform that action. If it's a
+nested menu, then we'll open the menu to that entry.
+
+Additionally, how exactly does that play with something like `{ "action":
+"newTab", "index": 1, "commandline": "wsl.exe" }`? This is really a discussion
+for that spec, but is an issue highlighted by this spec. If the first entry is
+anything other than a `profile`, then the `commandline` parameter doesn't really
+mean anything anymore. I'm tempted to revert this particular portion of the new
+tab menu customization spec over this.
+
+We could instead add an `index` to `openNewTabDropdown`, and have that string
+instead be "Open new tab dropdown, index:1". That would help disambiguate the
+two.
+
+## Future considerations
+
+* Some of these command names are starting to get _very_ long. Perhaps we need a
+  netting to display Command Palette entries on two lines (or multiple, as
+  necessary).
+* When displaying the entries of a nested command to the user, should we display
+  a small label showing the name of the previous command? My gut says _yes_. In
+  the Proposal 1 example, pressing `ctrl+alt+e` to jump to "Split Pane..."
+  should probably show a small label that displays "Split Pane..." above the
+  list of nested commands.
+
+
+## Appendix 1: Name generation samples for `ShortcutAction`s
+
+### Current `ShortcutActions`
+
+* `CopyText`
+  - "Copy text"
+  - "Copy text as a single line"
+  - "Copy text without formatting"
+  - "Copy text as a single line without formatting"
+* `PasteText`
+  - "Paste text"
+* `OpenNewTabDropdown`
+  - "Open new tab dropdown"
+* `DuplicateTab`
+  - "Duplicate tab"
+* `NewTab`
+  - "Open a new tab, profile:{profile name}, directory:{directory}, commandline:{commandline}, title:{title}"
+* `NewWindow`
+  - "Open a new window"
+  - "Open a new window, profile:{profile name}, directory:{directory}, commandline:{commandline}, title:{title}"
+* `CloseWindow`
+  - "Close window"
+* `CloseTab`
+  - "Close tab"
+* `ClosePane`
+  - "Close pane"
+* `NextTab`
+  - "Switch to the next tab"
+* `PrevTab`
+  - "Switch to the previous tab"
+* `SplitPane`
+  - "Open a new pane, profile:{profile name}, split direction:{direction}, split size:{X%/Y chars}, resize parents, directory:{directory}, commandline:{commandline}, title:{title}"
+  - "Duplicate the current pane, split direction:{direction}, split size:{X%/Y chars}, resize parents, directory:{directory}, commandline:{commandline}, title:{title}"

```

Comment:
I hate this so much. It'll only get worse as we add more keybinding args to override aspects of a profile. Can we just do "Open a new pane with profile {profile name} (Custom)"?

And if there's multiple, we start adding numbers to "Custom".

Reply:
See, I wanted to ensure that whatever set of args the user provided always generated a unique string. Yea, if the user puts all the args into a keybinding/command, this will be stupid long like this. At that point, the user should probably have a better name they could give the command.